packages/core/src/bin/commands/prune.ts

```diff
@@ -0,0 +1,167 @@
+  const common = { options, logger, metrics, telemetry, shutdown };
+
+  const build = await createBuild({ common, cliOptions });
+
+  const exit = createExit({ common });
+
+  const configResult = await build.executeConfig();
+  if (configResult.status === "error") {
+    await exit({ reason: "Failed intial build", code: 1 });
+    return;
+  }
+
+  const buildResult = build.preCompile(configResult.result);
+
+  if (buildResult.status === "error") {
+    await exit({ reason: "Failed intial build", code: 1 });
+    return;
+  }
+
+  const database = await createDatabase({
+    common,
+    // Note: `namespace` is not used in this command
+    namespace: "public",
+    preBuild: buildResult.result,
+    schemaBuild: emptySchemaBuild,
+  });
+
+  const TABLES = pgSchema("information_schema").table("tables", (t) => ({
+    table_name: t.text().notNull(),
+    table_schema: t.text().notNull(),
+  }));
+
+  const ponderSchemas = await database.qb.drizzle
+    .select({ schema: TABLES.table_schema, tableCount: count() })
+    .from(TABLES)
+    .where(
+      inArray(
+        TABLES.table_schema,
+        database.qb.drizzle
+          .select({ schema: TABLES.table_schema })
+          .from(TABLES)
+          .where(eq(TABLES.table_name, "_ponder_meta")),
+      ),
+    )
+    .groupBy(TABLES.table_schema);
+
+  const queries = ponderSchemas.map((row) =>
+    database.qb.drizzle
+      .select({
+        value: getPonderMeta(row.schema).value,
+        schema: sql<string>`${row.schema}`.as("schema"),
+      })
+      .from(getPonderMeta(row.schema))
+      .where(eq(getPonderMeta(row.schema).key, "app")),
+  );
+
+  if (queries.length === 0) {
+    console.log("No inactive Ponder apps found in this database.\n");
+    await exit({ reason: "Success", code: 0 });
+    return;
+  }
+
+  let result: { value: PonderApp; schema: string }[];
+
+  if (queries.length === 1) {
+    result = await queries[0]!;
+  } else {
+    // @ts-ignore
+    result = await unionAll(...queries);
+  }
+
+  const tablesToDrop: string[] = [];
+  const schemasToDrop: string[] = [];
+  const functionsToDrop: string[] = [];
+
+  for (const { value, schema } of result) {
+    if (value.is_dev === 1) continue;
+    if (
+      value.is_locked === 1 &&
+      value.heartbeat_at + common.options.databaseHeartbeatTimeout > Date.now()
+    ) {
+      continue;
+    }
+
+    for (const table of value.table_names) {
+      tablesToDrop.push(`"${schema}"."${table}"`);
+      tablesToDrop.push(`"${schema}"."${sqlToReorgTableName(table)}"`);
+      functionsToDrop.push(`"${schema}"."operation_reorg__${table}"`);
+    }
+    tablesToDrop.push(`"${schema}"."_ponder_meta"`);
+    tablesToDrop.push(`"${schema}"."_ponder_status"`);
+
+    const tableCount = ponderSchemas.find(
+      (s) => s.schema === schema,
+    )!.tableCount;
+
+    if (schema !== "public" && tableCount <= 2 + value.table_names.length * 2) {
+      schemasToDrop.push(schema);
+    }
```

Comment:
What's the logic here? Are you hesitant to simply do DROP SCHEMA CASCADE here?

Reply:
The schema is only dropped if we can be pretty sure there are no other tables besides Ponder tables. 2 + x * 2 is the formula used to account for reorg tables, _ponder_meta and _ponder_status